### PR TITLE
Generalize onboarding process

### PR DESCRIPTION
**Pull request by lguychard**
As discussed during the product/eng manager sync, lifts up the new web onboarding process to apply to all new engineering hires.

### CODE REVIEW

File: handbook/engineering/onboarding.md [hunk 1]
```diff
@@ -15,29 +40,62 @@ Welcome to Sourcegraph! This document will guide you through engineering specifi
...
-- [Complete general onboarding](../people-ops/onboarding.md#for-all-new-teammates)
-- [Configure your GitHub notifications.](github-notifications/index.md)
-- Message each person on your immediate team to setup a time to meet them. Some things you might want to learn:
-  - What brought them to Sourcegraph?
-  - What are they currently working on?
-  - What do they do for fun?
-- Join #dev-announce, #dev-chat, and your team's channel on Slack. [Team chat documentation](../communication/team_chat.md#engineering)
+Your manager will assign to you three starter tasks that you should aim to complete in your first three weeks. These tasks are small, atomic, and intended to expose you to different parts of our codebase and product: it's important that you build the flexibility to fix any problem you'll be faced with at Sourcegraph, and don't narrow down your comfort zone to a certain part of our codebase or product.
+
+As you're working on these tasks:
```
Comment (mrnugget) [inline handbook/engineering/onboarding.md:50]: When I joined @beyang said to me: "In your first few weeks: optimise for learning" @nicksnyder said something along the same lines and it really stuck with me. It takes a lot of the "i need to ship asap" pressure of when starting.

Maybe we can add something like this to this section?

File: handbook/engineering/onboarding.md [hunk 2]
```diff
@@ -1,9 +1,34 @@
...
+At Sourcegraph, we don't think of engineers as resources — we think of them as owners of their work, who constantly reevaluate how to use their talents to be as impactful as possible. We value your opinions and ideas. You should always feel empowered to identify potential improvements and act upon them, whether they be improvements to this onboarding process, our handbook and general documentation, our codebase and tooling, or our product.
+
+Never assume that a problem is somebody else's to fix!
+
+## Getting set up
+
+You'll have to get some basics set up in your first few days:
+- Complete [general onboarding](../people-ops/from-graphbook/onboarding.md#for-all-new-teammates)
+- [Configure your GitHub notifications.](./github-notifications/index.md)
+- Join #dev-announce, #dev-chat, and your team's channel on Slack. [Team chat documentation](../communication/team_chat.md#engineering)
```
Comment (tsenart) [inline handbook/engineering/onboarding.md:24]:
```suggestion
- Join #dev-announce, #dev-chat, and your team's channel on Slack, as well as any other channels you find interesting. [Team chat documentation](../communication/team_chat.md#engineering)
```

File: handbook/engineering/onboarding.md [hunk 3]
```diff
@@ -15,29 +40,62 @@ Welcome to Sourcegraph! This document will guide you through engineering specifi
...
+- Discuss your plan with the team and your manager.
+- Get hacking!
+
+At Sourcegraph, you'll be expected to own the problems your team is solving, and work with the team to define solutions and plan your work — we think it's important that you start doing this early on!
+
+### Give feedback on your onboarding
+
+At the end of week 6 at Sourcegraph, your manager will send you a survey to learn more about what worked and what didn't during your onboarding. Take your time to answer this thoughtfully — your answers will be very important to make sure our onboarding process is even better for future hires!
+
+TODO @lguychard: prepare this
```
Comment (tsenart) [inline handbook/engineering/onboarding.md:101]:
```suggestion
```

File: handbook/engineering/onboarding.md [hunk 4]
```diff
@@ -1,9 +1,34 @@
...
 
+### There are no stupid questions
+
+Joining a new company can be overwhelming — there's a lot to learn! As you navigate your first few weeks at Sourcegraph, we want you to know that everyone on the team is here to help, and that there are **no stupid questions**.
+
+Every time you're curious or confused about something — just ask! When you do so, use [public discussion channels](../communication/team_chat.md#avoid_private_messages) as much as possible.
+
+### Think and act like an owner
+
+At Sourcegraph, we don't think of engineers as resources — we think of them as owners of their work, who constantly reevaluate how to use their talents to be as impactful as possible. We value your opinions and ideas. You should always feel empowered to identify potential improvements and act upon them, whether they be improvements to this onboarding process, our handbook and general documentation, our codebase and tooling, or our product.
```
Comment (tsenart) [inline handbook/engineering/onboarding.md:15]:
```suggestion
At Sourcegraph, we don't think of engineers as resources — we think of them as owners of their work, who constantly reevaluate how to use their talents to be as impactful as possible. We value your opinions and ideas. You should always feel empowered to identify potential improvements and act upon them, whether they be improvements to processes (like onboarding), our handbook and general documentation, our codebase and tooling, or our product.
```

File: handbook/engineering/onboarding.md [hunk 5]
```diff
@@ -1,9 +1,34 @@
...
+### Think and act like an owner
+
+At Sourcegraph, we don't think of engineers as resources — we think of them as owners of their work, who constantly reevaluate how to use their talents to be as impactful as possible. We value your opinions and ideas. You should always feel empowered to identify potential improvements and act upon them, whether they be improvements to processes (like onboarding), our handbook and general documentation, our codebase and tooling, or our product.
+
+Never assume that a problem is somebody else's to fix!
+
+## Getting set up
+
+You'll have to get some basics set up in your first few days:
+- Complete [general onboarding](../people-ops/from-graphbook/onboarding.md#for-all-new-teammates)
```
Comment (lguychard) [inline handbook/engineering/onboarding.md:22]:
```suggestion
- Complete [general onboarding](../people-ops/onboarding/index.md#for-all-new-teammates)
```

File: handbook/engineering/onboarding.md (hunk 1 above)
Comment (lguychard) [reply handbook/engineering/onboarding.md:50]: What do you think of this?

```suggestion
As you're working on these tasks:
- Optimize for learning: your priority is to soak up as much context as possible.
```

Comment (mrnugget) [reply handbook/engineering/onboarding.md:50]: Yep, that's good, or go even further:

```suggestion
As you're working on these tasks:
- Optimize for learning: your priority is to soak up as much context as possible, not to ship something as fast as possible.
```

I'll let you decide, though.

**Pull request merged**